docs/templates/Object Definition/Property Sets/Property Sets for Materials/README.md

```diff
@@ -0,0 +1,31 @@
+Property Sets for Materials
+===========================
+
+The concept template _Property Sets for Materials_ describes how a material can be related to a single or multiple property sets. A property set contains a single or multiple properties. The data types of an individual property are single value, enumerated value, bounded value, table value, reference value, list value, and combination of property occurrences.
+
+NOTE: An IfcMaterial is not an IfcObject. For legacy reasons, this concept is listed under _Object Definition_. In future versions of the standard the various property set association mechanisms will likely be unified. In the current version of this template, there is no mechanism to further guide the applicability in a way similar to how applicability can be tailored to individual predefined types of IfcObject. In future versions of the standard there will likely be the possibility to make property sets applicable to specific categories of IfcMaterial.
+
+Property Sets applicable to this template will carry the IfcPropertySetTemplateTypeEnum "PSET_MATERIALDRIVEN".
+
+```
+concept {
+    IfcMaterial:HasProperties -> IfcMaterialProperties:Material
```

Comment:
I guess you resolve inherited properties? I am assuming the first line also defines the applicable entity. I would prefer IfcMaterialDefinition:HasProperties honestly.

Reply:
> I am assuming the first line also defines the applicable entity

By convention yes, but it's actually determined as the entity without any incoming edges. (incoming edges in the -> arrow notation, the edge in the rendered diagram will be reversed as it is an inverse attribute).

> I would prefer IfcMaterialDefinition:HasProperties honestly.

Me too actually. IfcMaterial was mostly used because we might want to use Category later as we use PredefinedType, but since we do not do that now, I agree with you.

Reply:
IfcMaterialDefinition is necessary because properties may also be assigned to layers, constituents, profiles, and the sets themselves. I always found that a bit of a "whoah" moment when implementing but that's just what it is right now.

Reply:
Well, it appears we are in agreement, I'm really not sure though if associating to Layers and Constituents is generally a good idea. They're just wrappers around the material to give some context on the location within the element. It'd introduce confusion to constantly need to check if the property set is defined on layer on or on the material itself.

Also, I'm not too familiar with IfcMaterialProfile. Do we still need a concept "Property Sets for Profiles" if there is already the possibility to wrap an IfcProfileDef in a IfcMaterialProfile?

Reply:
@aothms profiles are an interesting thing in that especially from a structural perspective it is important for them to be identified separately. A structural engineer will have an IfcProjectLibrary of profiles, which then they may choose to combine into IfcMaterialProfiles. See for example Pset_ProfileMechanical - I think @Jesusbill has helped look into this in the past with me.